enclave/src/lib.rs

```diff
@@ -41,8 +45,11 @@ pub extern "C" fn ecall_test(some_string: *const u8, some_len: usize) -> sgx_sta
     sgx_status_t::SGX_SUCCESS
 }
 
+/// # Safety
+/// The caller needs to ensure that `binary` is a valid pointer to a slice valid for `binary_len` items
+/// and that `result_out` is a valid pointer.
 #[no_mangle]
-pub extern "C" fn exec_wasm_test(
+pub unsafe extern "C" fn exec_wasm_test(
```

Comment:
Consider removing the `unsafe` keyword

Reply:
So the reason I have an unsafe function and an unsafe block is to make it clear that this function puts certain expectations on the caller with regards to the pointers being dereferenced. This is something picked up by clippy: https://rust-lang.github.io/rust-clippy/master/index.html#not_unsafe_ptr_arg_deref

I am also using this feature https://rust-lang.github.io/rfcs/2585-unsafe-block-in-unsafe-fn.html. This allows us to be explicit about the unsafe parts of an unsafe function as well.